R/link_to_nhd.R

```diff
@@ -56,5 +82,10 @@ link_to_waterbodies = function(lats, lons, ids){
 
   unique_matches = unique(do.call(rbind, match_res))
   #return matches that have non-NA value PREMANENT_ID
-  return(unique_matches[!is.na(unique_matches$PERMANENT_), ])
+  if(tolower(dataset) == "nhd"){
+    return(unique_matches[!is.na(unique_matches$PERMANENT_), ])
+  }
+  else if(tolower(dataset) == "hydrolakes"){
+    return(unique_matches[!is.na(unique_matches$Hylak_id), ])
```

Comment:
Maybe the ID name could be defined in that same overarching if/else at the top of the function?